### PR TITLE
cmake: add new versions + remove old ones + use official src downloads

### PR DESCRIPTION
**Pull request by Croydon**
* add new versions
* remove old versions
* use official source downloads for all versions; some used the automated GitHub generated source archives
* some additional tiny improvements

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Croydon** [comment]
The test_package should be perfectly fine for both 1.x and 2.x. Do I really have to create an identical `test_v1_package`?

File: recipes/cmake/3.x.x/conanfile.py [hunk 1]
```diff
@@ -125,15 +122,15 @@ def package_id(self):
...
+        module_version = "{}.{}".format(Version(self.version).major, Version(self.version).minor)
 
         bindir = os.path.join(self.package_folder, "bin")
         self.output.info("Appending PATH environment variable: {}".format(bindir))
         self.env_info.PATH.append(bindir)
 
         self.buildenv_info.prepend_path("CMAKE_ROOT", self.package_folder)
         self.env_info.CMAKE_ROOT = self.package_folder
-        mod_path = os.path.join(self.package_folder, "share", "cmake-%s" % minor, "Modules")
+        mod_path = os.path.join(self.package_folder, "share", "cmake-%s" % module_version, "Modules")
```
Comment (jwillikers) [inline recipes/cmake/3.x.x/conanfile.py:133]:
```suggestion
        mod_path = os.path.join(self.package_folder, "share", f"cmake-{module_version}", "Modules")
```

File: recipes/cmake/3.x.x/test_package/conanfile.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
 import os
 from six import StringIO
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.build import can_run
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch"
```
Comment (jwillikers) [inline recipes/cmake/3.x.x/test_package/conanfile.py:8]:
```suggestion
    settings = "os", "arch"
    test_type = "explicit"
    generators = "VirtualRunEnv"
```

File: recipes/cmake/3.x.x/test_package/conanfile.py [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
     def test(self):
-        if not tools.cross_building(self):
+        if can_run(self):
             output = StringIO()
             self.run("cmake --version", output=output, run_environment=True)
```
Comment (jwillikers) [inline recipes/cmake/3.x.x/test_package/conanfile.py:16]:
```suggestion
            self.run("cmake --version", env="conanrun", output=output)
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/cmake/3.x.x/test_package/conanfile.py (hunk 2 above)
Comment (Croydon) [reply recipes/cmake/3.x.x/test_package/conanfile.py:8]: The `test_type` attribute is getting ignored in 2.x so it probably should be added to the 1.x test_package instead

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**